astropy/cosmology/scalar_inv_efuncs.pyx

```diff
@@ -0,0 +1,225 @@
+""" Cython inverse efuncs for cosmology integrals"""
+#cython boundcheck=False
```

Comment:
Yeah, if you look in the C code, there's quite a bit of mucking around with accessing the array via the buffer interface, that even with boundscheck that setup takes longer than actually doing the calculation itself for N < 4.  But you're really getting into the weeds at that point anyways.

The actual fastest approach I found was to have two versions of the function: One that is hard-coded to take 3 neutrino species and has the loop unrolled, and one that can take an arbitrary number, and uses a loop.  But it's really overkill.